modules/ocf/files/packages/cups/convert_failure

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+"""A script to send an email to the root mailing list upon failure of
+the convert program, when used in the raster-filter script, a
+Tea4CUPS filter.
+
+Jobs printed from one of the desktops, or from supernova or tsunami,
+first go through raster-filter, which calls convert to reduce the
+density of any PDFs, essentially rasterizing them. If convert happens
+to fail, raster-filter will pipe stderr from convert to this script,
+which takes in stdin and includes it in an email to root@ocf. Thus,
+this script should not be called casually by regular users!
+
+Based off the enforcer script for whiteout, and the
+email_ldapsync_problem script
+"""
+import os
+import socket
+import sys
+from textwrap import dedent
+
+from ocflib.misc import mail
+
+ERROR_MSG = dedent("""\
+        raster-filter encountered the following error while running convert:
+
+        {stdin}
+
+
+        tea4cups environment variables:
+        {tea_vars}
+
+        ====
+        Hostname: {hostname}
+        """.format(
+    stdin=sys.stdin.read(),
```

Comment:
The `sys.stdin.read()` call is blocking, which could cause problems in this script. I'd fill in this template within the main function.

Reply:
Thanks for the suggestion! I will fill in my `ERROR_MSG` placeholders within main.

To clarify, when you say that the `sys.stdin.read()` call is blocking, can you explain what you mean?

Reply:
It performs an IO operation that isn't guaranteed to return instantly-- if `stdin` isn't filled up, the rest of the program will never run.

It also falls into the realm of program logic, which you want to make sure stays in the main function (or is called from the main function). Constants can be declared at the top level, but this isn't a constant. See https://stackoverflow.com/a/1954796 for some more explanation on why we do this.